src/Streamly/Fold.hs

```diff
@@ -952,14 +950,21 @@ ltakeWhile predicate (Fold step initial done) = Fold step' initial' done'
 -- prefix of length @n@ and the second group consisting of the rest of the
 -- stream.
 --
+-}
+{-# INLINE splitAt #-}
 splitAt
     :: Monad m
     => Int
     -> Fold m a b
     -> Fold m a c
     -> Fold m a (b, c)
-splitAt n = undefined
--}
+splitAt n (Fold stepL initL doneL) (Fold stepR initR doneR) = Fold step init done
+  where
+    step (ind,v1,v2) input =
+            if ind > 0 then stepL v1 input >>= (\a -> return (ind-1,a,v2))
```

Comment:
We use the following style for if-then-else:

```
if x
then y
else x
```

Also, always put spaces around operators and after ",", e.g.

```
(ind - 1, a, v2)
```

Use `xL` and `xR` instead of `v1` and `v2`. That provides a hint about which stream they belong to.